tests/test_e2e/test_pebble.py

```diff
@@ -437,4 +439,46 @@ def _on_custom_notice(self, event: PebbleCustomNoticeEvent):
     )
     state = State(containers=[container])
     ctx = Context(MyCharm, meta={"name": "foo", "containers": {"foo": {}}})
-    ctx.run(container.get_notice(key).event, state)
+    ctx.run(ctx.on.pebble_custom_notice(container=container, notice=notices[-1]), state)
+
+
+def test_pebble_check_failed():
+    class MyCharm(CharmBase):
+        def __init__(self, framework):
+            super().__init__(framework)
+            framework.observe(self.on.foo_pebble_check_failed, self._on_check_failed)
+
+        def _on_check_failed(self, event):
+            info = event.info
+            assert info.name == "http-check"
+            assert info.status == pebble.CheckStatus.DOWN
+            assert info.failures == 7
+
+    ctx = Context(MyCharm, meta={"name": "foo", "containers": {"foo": {}}})
+    check = Check("http-check", failures=7, status=pebble.CheckStatus.DOWN)
+    container = Container("foo", checks={check})
+    state = State(containers={container})
+    ctx.run(ctx.on.pebble_check_failed(check=check, container=container), state=state)
```

Comment:
I see the asserts in `_on_check_failed`, but this doesn't actually ensure that that methods ran at all. Should we do something like:

```
def test_pebble_check_failed():
    event_ran = False
    class MyCharm(CharmBase):
        ...
        def _on_check_failed(self, event):
            nonlocal event_ran
            event_ran = True
            ...

   ...
   ctx.run(ctx.on.pebble_check_failed(...), state=state)
   assert event_ran
```